### PR TITLE
Add method to permit creating `BackgroundField`s out of `Number`s directly

### PR DESCRIPTION
**Pull request by tomchor**
See https://github.com/CliMA/Oceananigans.jl/pull/3813

### CODE REVIEW

**Comment by glwagner** [comment]
Nice. It's not tested until we use that example though. We could also go the extra step of testing it directly if we are going to add it separately>?

**Comment by glwagner** [comment]
happy to merge this as is and then test it with the example

**Review by glwagner — APPROVED** [review]

**Comment by tomchor** [comment]
I think we should test this independently of an example. Examples change but it'd be nice to make sure this functionality still works. I'll add something simple and then we can iterate.

File: src/Fields/background_fields.jl [hunk 1]
```diff
@@ -60,6 +60,8 @@ func(x, y, z, t, parameters)
 """
 BackgroundField(func; parameters=nothing) = BackgroundField(func, parameters)
 
+regularize_background_field(LX, LY, LZ, c::Number, grid, clock) = ConstantField(c)
```
Comment (glwagner) [inline src/Fields/background_fields.jl:63]:
```suggestion
regularize_background_field(LX, LY, LZ, bg::BackgroundField{<:Number}, grid, clock) = ConstantField(bg.func)
```

My b, this is what it should look like

I guess we assumed the background fields would be functions and called that property `func`...

I'd chalk up that cleanup to also resolving #3586 not worth tackling here

**Pull request merged**